dhcpv6/ztpv6/parse_vendor_options.go

```diff
@@ -66,6 +69,27 @@ func ParseVendorData(packet dhcpv6.DHCPv6) (*VendorData, error) {
 			vd.Model = p[1]
 			vd.Serial = p[2]
 			return &vd, nil
+
+		// For Ciena the class identifier (opt 60) is written in the following format:
+		//    {vendor iana code}-{product}-{type}
+		// For Ciena the iana code is 1271
+		// The product type is a number that maps to a Ciena product
+		// The type is used to identified different subtype of the product.
+		// An example can be ‘1271-23422Z11-123’.
+		case strings.HasPrefix(d, strconv.Itoa(int(iana.EnterpriseIDCienaCorporation))):
+			v := strings.Split(d, "-")
+			if len(v) < 3 {
+				return nil, errVendorOptionMalformed
+			}
+			duid := packet.(*dhcpv6.Message).Options.ClientID()
+			if duid.Type != dhcpv6.DUID_EN {
```

Comment:
Is this check actually needed? Or is it only to provide additional certainty?

Reply:
Specifications from the vendor is that this parameter is always DUID_EN, this is why I have enforced it here, I'm also not sure what is going to happen if type != DUID_EN and I try to access duid.EnterpriseIdentifier ?